### PR TITLE
Gateway and Network functionalities

### PR DESCRIPTION
**Pull request by RohanShrothrium**
The PR introduces basic functionalities of Network and Gateway which will be used to implement other higher level APIs

### CODE REVIEW

File: hfc/fabric_network/gateway.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import logging
```
Comment (dexhunter) [inline hfc/fabric_network/gateway.py]: Hi! Could you also add a line (`# SPDX-License-Identifier: Apache-2.0
`) at the top? Thanks!

Comment (RohanShrothrium) [reply hfc/fabric_network/gateway.py]: Will add the test cases in the future step.

**Comment by yeasy** [comment]
TODO: test cases and tutorial of the API usage.

**Comment by RohanShrothrium** [comment]
> TODO: test cases and tutorial/docs for the API usage.



> TODO: test cases and tutorial/docs for the API usage.

Will start working on this!

**Comment by ryjones** [comment]
please rebase to the tip of master

**Comment by RohanShrothrium** [comment]
@ryjones 
The branch is already up to date!

**Pull request merged**